plugin/RunalyzePluginPanel_TagsSummary/class.RunalyzePluginPanel_TagsSummary.php

```diff
@@ -140,14 +141,16 @@ protected function displayContent() {
 	/**
 	 * @param \Runalyze\Model\Sport\Entity $Sport
 	 */
-	protected function showListFor(Model\Sport\Entity $Sport) {
+	protected function showListFor(Model\Sport\Entity $Sport, $timescale) {
 		$max = 0;
 		$countTagsForSport = DB::getInstance()
-			->query('SELECT '.PREFIX.'activity_tag.tagid, '.PREFIX.'tag.tag, COUNT(tagid) as `count`'.
-				' FROM '.PREFIX.'activity_tag'.
-				' LEFT JOIN '.PREFIX.'training ON '.PREFIX.'training.sportid = '.$Sport->id().
-				' LEFT JOIN '.PREFIX.'tag ON '.PREFIX.'tag.id = '.PREFIX.'activity_tag.tagid'.
-				' WHERE '.PREFIX.'training.id = activityid GROUP BY tagid ORDER BY count DESC');
+			->query('SELECT activity_tag.tagid, tag.tag, COUNT(tagid) as `count`'.
+				' FROM '.PREFIX.'activity_tag activity_tag'.
+				' LEFT JOIN '.PREFIX.'training training ON training.sportid = '.$Sport->id().
+				' LEFT JOIN '.PREFIX.'tag tag ON tag.id = activity_tag.tagid'.
+				' WHERE training.id = activityid'.
+				' AND FROM_UNIXTIME(training.time-IFNULL(training.timezone_offset,0)*60) >= NOW()-INTERVAL '.(int)$timescale.' MONTH'.
```

Comment:
Activity timestamps are treated as if the user's time zone is UTC and MySQL's time zone is set to UTC as well, so `FROM_UNIXTIME(training.time) >= NOW() - INTERVAL '.(int)$timescale.' MONTH` should suffice. `timezone_offset` is only an informational column for the activity view but is not of interest for handling timestamps.